### PR TITLE
[Breaking Change] Several improvements - Added Async Support

### PR DESCRIPTION
**Pull request by XChikuX**
Changes Include:
- Added UV package management
- Added Tox tests from py310 -> py313
- Added Asynchronous Support
- Added Batching support
- Added More Advanced Queries
- Removed dependency on `dotenv` which can lead to security issues
- Revv to version 1.8
- Added myself as a maintainer

**NOTE:** You must add connection parameters to the environment manually. Either through python or through the command line.

I recommend using [Doppler](https://www.doppler.com/)

Remaining Work:
- Add step in CI to auto commit to pypi once all tests pass.

### CODE REVIEW

**Comment by XChikuX** [comment]
@Marcus-Peterson Would love to get these in. Please feel free perform checks on your end :)

**Comment by XChikuX** [comment]
Fixes #1

**Comment by Marcus-Peterson** [comment]
Holy shit, you did a lot. Uhh, let me uh... Review this later today... 

Currently sitting on the toilet while reviewing your commits, so can't go deep into the additional functionalities you made. But! From the looks of it. Looks promising.

**Comment by Marcus-Peterson** [comment]
Btw, you have tested this, thoroughly, right..?

**Comment by XChikuX** [comment]
> Btw, you have tested this, thoroughly, right..?

We can benefit from a little more testing. Don't approve these just yet.
I'd like if you can go over the changes and let me know if you are comfortable with everything that is suggested, though. :)

If we disagree on something major, I'll have to re-adjust what is appropriate for this repo.

File: .github/workflows/python-publish.yml [hunk 1]
```diff
@@ -51,11 +57,11 @@ jobs:
     environment:
       name: pypi
       # OPTIONAL: uncomment and update to include your PyPI project URL in the deployment status:
-      # url: https://pypi.org/p/YOURPROJECT
+      # url: https://pypi.org/p/tursopy
```
Comment (Marcus-Peterson) [inline .github/workflows/python-publish.yml]: This is often something that is confused... "tursopy" is actually the name of another project that tries to acheive basically the same thing. The correct name for this package is:
https://pypi.org/project/turso-python/

So it should be "turso-python" and not "tursopy"

File: .env
Comment (Marcus-Peterson) [inline .env:1]: LGTM

File: .github/workflows/ci.yml
Comment (Marcus-Peterson) [inline .github/workflows/ci.yml:1]: LGTM

File: .github/workflows/python-publish.yml
Comment (Marcus-Peterson) [inline .github/workflows/python-publish.yml:1]: LGTM

File: .gitignore
Comment (Marcus-Peterson) [inline .gitignore:1]: LGTM

File: README.md
Comment (Marcus-Peterson) [inline README.md:1]: If you guys have the time, add usage of the verify_integration.py from the scripts folder into the readme. Other than that: LGTM

File: pyproject.toml
Comment (Marcus-Peterson) [inline pyproject.toml:1]: LGTM

File: pytest.ini
Comment (Marcus-Peterson) [inline pytest.ini:1]: LGTM

File: requirements.txt
Comment (Marcus-Peterson) [inline requirements.txt:1]: I dont like that you removed this, but who cares. LGTM

File: scripts/async_crud_smoke_test.py
Comment (Marcus-Peterson) [inline scripts/async_crud_smoke_test.py:1]: All relevant edge cases seems to have been covered except for retry timeout, ratelimit and test for when input is incorrect. Other than that: LGTM

File: scripts/crud_smoke_test.py
Comment (Marcus-Peterson) [inline scripts/crud_smoke_test.py:1]: LGTM, the sync version doesn't have rate limit or retry timeout so no need to test that. Could potentially be valuable something to add in the future.

File: scripts/verify_integration.py
Comment (Marcus-Peterson) [inline scripts/verify_integration.py:1]: LGTM

File: setup.py
Comment (Marcus-Peterson) [inline setup.py:1]: LGTM

File: scripts/crud_smoke_test.py
Comment (XChikuX) [reply scripts/crud_smoke_test.py:1]: It's a little risky to add retries to something that runs synchronously. It would work well with something like FastAPI since it automatically gets converted to a thread.

But not all frameworks do that. I'd cover this if it weren't for that.

Optimal right now if it just fails and user manually implements retries from the frontend, which they likely do.

On an async backend, retries don't take up any blocking space ever. So it works out well.

File: test.py
Comment (Marcus-Peterson) [inline test.py:1]: LGTM

File: tests/test_async.py
Comment (Marcus-Peterson) [inline tests/test_async.py:1]: LGTM

File: tests/test_async_foreign_key.py
Comment (Marcus-Peterson) [inline tests/test_async_foreign_key.py:1]: LGTM

File: scripts/async_crud_smoke_test.py
Comment (Marcus-Peterson) [inline scripts/async_crud_smoke_test.py:1]: LGTM

File: turso_python/__init__.py
Comment (Marcus-Peterson) [inline turso_python/__init__.py:1]: LGTM

File: turso_python/async_connection.py [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+    return url
+
+
+class AsyncTursoConnection:
+    def __init__(
+        self,
+        database_url: str | None = None,
+        auth_token: str | None = None,
+        *,
+        timeout: int = 30,
```
Comment (Marcus-Peterson) [inline turso_python/async_connection.py:27]: I'm unfamiliar with how aiohttp handle timeout exceptions, should this specific excpetion be handled on the client side or interface side?

File: turso_python/async_crud.py
Comment (Marcus-Peterson) [inline turso_python/async_crud.py:1]: LGTM

File: turso_python/batch.py
Comment (Marcus-Peterson) [inline turso_python/batch.py:1]: LGTM

File: turso_python/connection.py
Comment (Marcus-Peterson) [inline turso_python/connection.py:1]: LGTM

File: turso_python/crud.py
Comment (Marcus-Peterson) [inline turso_python/crud.py:1]: LGTM

File: turso_python/exceptions.py
Comment (Marcus-Peterson) [inline turso_python/exceptions.py:1]: LGTM

File: turso_python/response_parser.py
Comment (Marcus-Peterson) [inline turso_python/response_parser.py:1]: LGTM

File: turso_python/result.py
Comment (Marcus-Peterson) [inline turso_python/result.py:1]: LGTM

File: turso_python/schema_validator.py
Comment (Marcus-Peterson) [inline turso_python/schema_validator.py:1]: LGTM

File: turso_python/turso_vector.py
Comment (Marcus-Peterson) [inline turso_python/turso_vector.py:1]: LGTM

**Review by Marcus-Peterson — COMMENTED** [review]
Finished reviewing everything, waiting on proof-of-tests, after that I will approve merge

File: turso_python/async_connection.py (hunk 2 above)
Comment (XChikuX) [reply turso_python/async_connection.py:27]: Each server-side library needs to wait a small amount of time to retrieve the data. If this data is not given in a reasonable amount of time. Then a timeout exception must be raised.

The client can then decide whether it wants to send a retry or not.

The client side timeout/retry is independent of how the server handles these timeouts. However, a 30 second timeout is a bit excessive; I'll chop it down to 3-5 seconds.

**Comment by Marcus-Peterson** [comment]
It's been a while guys, are you ready for a merge?

**Comment by Marcus-Peterson** [comment]
I'm just gonna merge this. Leeroy Jenkins
